/* -------------------------------------------------------------------------- */
/* Copyright 2002-2025, OpenNebula Project, OpenNebula Systems                */
/*                                                                            */
/* Licensed under the Apache License, Version 2.0 (the "License"); you may    */
/* not use this file except in compliance with the License. You may obtain    */
/* a copy of the License at                                                   */
/*                                                                            */
/* http://www.apache.org/licenses/LICENSE-2.0                                 */
/*                                                                            */
/* Unless required by applicable law or agreed to in writing, software        */
/* distributed under the License is distributed on an "AS IS" BASIS,          */
/* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   */
/* See the License for the specific language governing permissions and        */
/* limitations under the License.                                             */
/* -------------------------------------------------------------------------- */
syntax = "proto3";

import "shared.proto";

package one.system;

option go_package = "github.com/OpenNebula/one/src/oca/go/src/goca/api/system";

message VersionRequest
{
    string session_id = 1;
}

message ConfigRequest
{
    string session_id = 1;
}

message SqlRequest
{
    string session_id = 1;
    string sql        = 2;
    bool federate     = 3;
}

message SqlQueryRequest
{
    string session_id = 1;
    string sql        = 2;
}

service SystemService
{
  rpc Version (one.system.VersionRequest) returns (one.ResponseXML);

  rpc Config (one.system.ConfigRequest) returns (one.ResponseXML);

  rpc Sql (one.system.SqlRequest) returns (one.ResponseID);

  rpc SqlQuery (one.system.SqlQueryRequest) returns (one.ResponseXML);
}
